crates/turbo-tasks/src/manager.rs

```diff
@@ -317,7 +317,8 @@ impl<B: Backend> TurboTasks<B> {
 
     /// Call a native function with arguments.
     /// All inputs must be resolved.
-    pub(crate) fn native_call(&self, func: FunctionId, inputs: Vec<TaskInput>) -> RawVc {
+    pub(crate) fn native_call(&self, func: FunctionId, mut inputs: Vec<TaskInput>) -> RawVc {
+        inputs.shrink_to_fit();
```

Comment:
Please move both this call and the two others to `PersistentTaskType::{native,resolve_native,resolve_trait},` constructors, with a single comment somewhere in the impl that explains why we need to save memory.

Reply:
I moved it into memory_backend to only shrink it when it's really stored into the task cache. So for a simple lookup it doesn't have to shrink it as it's dropped anyway.